prow/gerrit/client/client.go

```diff
@@ -179,14 +179,26 @@ func (l LastSyncState) DeepCopy() LastSyncState {
 	return result
 }
 
+type roundTripperWithHeader struct {
+	upstream http.RoundTripper
+}
+
+func (rt *roundTripperWithHeader) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Add("user-agent", "prow")
```

Comment:
Is `Header` guaranteed to be non-nil?

Reply:
yeah I believe so. This is common among all other examples I saw